components/Player/DeleteField.tsx

```diff
@@ -1,77 +1,56 @@
-import { useState } from "react";
-import { DeleteFieldDTO } from "pages/api/admin/users/player/delete";
-import { useRouter } from "next/router";
+import dayjs from "dayjs";
+import { IProfileField } from "interfaces/user";
+import { useContext, useState } from "react";
+import { deserializeProfileFieldValue } from "utils/buildUserProfile";
+import isAbsence from "utils/isAbsence";
+import labelProfileField from "utils/labelProfileField";
+import ProfileContext from "./ProfileContext";
 
-type Props = React.PropsWithChildren<{
-  setOption: React.Dispatch<React.SetStateAction<string>>;
-  setSuccess: React.Dispatch<React.SetStateAction<boolean>>;
-  setType: React.Dispatch<
-    React.SetStateAction<"updated" | "added" | "deleted" | undefined>
-  >;
-  setDate: React.Dispatch<React.SetStateAction<string>>;
-  id: number;
-  userId: number;
-  fieldType: string;
-  date: string;
-}>;
+type Props = {
+  field: IProfileField | Absence;
+  onComplete?: (deleted?: IProfileField | Absence) => void;
+};
 
-const DeleteField: React.FC<Props> = ({
-  setOption,
-  setSuccess,
-  setDate,
-  setType,
-  fieldType,
-  id,
-  userId,
-  date,
-}: Props) => {
-  const router = useRouter();
+const DeleteField: React.FC<Props> = ({ field, onComplete }: Props) => {
+  const { deleteField } = useContext(ProfileContext);
   const [error, setError] = useState("");
-  const link =
-    fieldType === "absence"
-      ? "/api/admin/users/player/deleteAbsences"
-      : "/api/admin/users/player/delete";
 
-  async function ScoreSubmit(event?: React.BaseSyntheticEvent): Promise<void> {
-    event?.preventDefault();
+  let date;
+  if (isAbsence(field)) {
+    date = field.date;
+  } else {
+    const value = deserializeProfileFieldValue(field);
+    if (typeof value === "object" && value && "date" in value) {
+      date = value.date;
+    } else {
+      date = field.createdAt;
+    }
+  }
+
+  async function onDelete(): Promise<void> {
     try {
-      const response = await fetch(link, {
-        method: "POST",
-        headers: { "Content-Type": "application/json" },
-        credentials: "include",
-        body: JSON.stringify({
-          id,
-          userId,
-        } as DeleteFieldDTO),
-      });
-      if (!response.ok) {
-        throw await response.json();
-      }
-      setDate(date);
-      setType("deleted");
-      setSuccess(true);
-      setOption("");
-      router.replace(router.asPath);
+      await deleteField(isAbsence(field) ? "absence" : field.key, field.id);
+      onComplete?.(field);
     } catch (err) {
       setError(err.message);
     }
   }
+
   return (
     <div>
       <p>
-        Are you sure you want to delete this{" "}
-        {fieldType === "Score" ? "Engagement Score" : ""}{" "}
-        {fieldType === "gpa" ? "GPA" : ""}{" "}
-        {fieldType === "absence" ? "Absence" : ""} for {date} ?
+        Are you sure you want to delete this {labelProfileField(field)} entry
+        for {dayjs(date).format("MM/DD/YYYY")} ?
```

Comment:
could we change this to MM/YYYY since I think there is actually no day that is tied to a ValueHistoryScore